### PR TITLE
Lifter cleaning

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterConstants.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package edu.greenblitz.robotName.subsystems.Lifter;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+
+public class LifterConstants {
+
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterConstants.java:8]: add backward fprward limit

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java [hunk 2]
```diff
@@ -48,8 +48,8 @@ public void setIdleMode(CANSparkMax.IdleMode idleMode) {
     }
 
     @Override
-    public void goToPosition(double position) {
-        motor.getPIDController().setReference(position, CANSparkMax.ControlType.kPosition, 0 , FEED_FORWARD.calculate(position));
+    public void goToPosition(Rotation2d position) {
+        motor.getPIDController().setReference(position.getRadians(), CANSparkMax.ControlType.kPosition, 0 , FEED_FORWARD.calculate(position.getRadians()));
```
Comment (YoniKiriaty) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java:52]: slot constant

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/Lifter.java [hunk 3]
```diff
@@ -53,12 +54,12 @@ public void setVoltage(double voltage) {
...
     }
 
-    public void resetEncoder(double position) {
+    public void resetEncoder(Rotation2d position) {
         lifter.resetEncoder(position);
     }
 
     public void resetEncoder() {
-        resetEncoder(NeoLifterConstants.ENCODER_POSE_WHEN_RESET);
+        resetEncoder(LifterConstants.ENCODER_POSE_WHEN_RESET);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/Lifter.java:62]: rename encoder pose to position

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/Lifter.java [hunk 4]
```diff
@@ -73,18 +74,18 @@ public boolean isSwitchPressed() {
         return lifterInputs.isBackwardSwitchPressed;
     }
 
-    public boolean isMotorAtPosition(double targetPosition) {
-        return Math.abs(targetPosition - lifterInputs.position) < NeoLifterConstants.TOLERANCE;
+    public boolean isMotorAtPosition(Rotation2d targetPosition) {
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/Lifter.java:77]:
```suggestion
      public boolean isAtPosition(Rotation2d targetPosition) {
```

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterConstants.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    public static final Translation3d ROBOT_RELATIVE_LIFTER_POSITION = new Translation3d(0,0.24,0.35);
+
+
+    public static final Rotation3d ROBOT_RELATIVE_LIFTER_ROTATION = new Rotation3d(Math.PI,0,-Math.PI/2);
+
+    public static final Rotation2d ENCODER_POSE_WHEN_RESET = Rotation2d.fromRotations(0);
+
+    public static final double TOLERANCE = 0.04;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterConstants.java:16]: remember that the tolerance is an angle. consider changing the tolerance to a rotation 2d

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java [hunk 6]
```diff
@@ -1,24 +1,26 @@
...
     private GBSparkMax motor;
 
     public NeoLifter() {
         motor = new GBSparkMax(NeoLifterConstants.MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
         motor.config(NeoLifterConstants.CONFIG);
 
-        motor.getReverseLimitSwitch(FORWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
-        motor.getForwardLimitSwitch(BACKWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
+        motor.getReverseLimitSwitch(NeoLifterConstants.FORWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
+        motor.getForwardLimitSwitch(NeoLifterConstants.BACKWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java:20]: for the enableLimitSwitch change the 'true' to a constant

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/SimulationLifter/SimulationLifterConstants.java [hunk 7]
```diff
@@ -12,11 +12,8 @@ public class SimulationLifterConstants {
     public static final double GEAR_RATIO = 30;
 
     public static final double MOMENT_OF_INERTIA = 0.03;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/SimulationLifter/SimulationLifterConstants.java:14]: use the         SingleJointedArmSim.estimateMOI()  function

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by yonnbro — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]
fix everything noam said tho

File: src/main/java/edu/greenblitz/robotName/commands/lifter/ReverseLifting.java [hunk 8]
```diff
@@ -1,21 +1,20 @@
...
     @Override
     public boolean isFinished() {
-        return lifter.isMotorAtPosition(NeoLifterConstants.LIFTER_RETRACTED_POSITION);
+        return lifter.isMotorAtPosition(LifterConstants.LIFTER_RETRACTED_POSITION);
     }
 
     @Override
     public void end(boolean interrupted) {
-        lifter.resetEncoder();
         lifter.stopMotor();
```
[the remark this answers is not in the document]
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/commands/lifter/ReverseLifting.java:18]: what?

[the remark this answers is not in the document]
Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/commands/lifter/ReverseLifting.java:18]: Noam was testing a hardware lock for the lifter, and I don't know if it made it to the final design... if it did we didn't add it to code, so please talk to him to make sure about it

**Review by aaa-png — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java [hunk 9]
```diff
@@ -1,24 +1,26 @@
...
+
     private GBSparkMax motor;
 
     public NeoLifter() {
         motor = new GBSparkMax(NeoLifterConstants.MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
         motor.config(NeoLifterConstants.CONFIG);
 
-        motor.getReverseLimitSwitch(FORWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
-        motor.getForwardLimitSwitch(BACKWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(true);
+        motor.getReverseLimitSwitch(NeoLifterConstants.FORWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(NeoLifterConstants.IS_BACKWARD_LIMIT_SWITCH_ENABLED);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java:19]: getReverseLimitSwitch recieves FORWARD_LIMIT_SWITCH_TYPE?

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java [hunk 10]
```diff
@@ -1,24 +1,26 @@
...
+        motor.getForwardLimitSwitch(NeoLifterConstants.BACKWARD_LIMIT_SWITCH_TYPE).enableLimitSwitch(NeoLifterConstants.IS_FORWARD_LIMIT_SWITCH_ENABLED);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/NeoLifter/NeoLifter.java:20]: the same?

**Review by tnrz135 — APPROVED** [review]

**Review by yonnbro — APPROVED** [review]

**Review by NoamRosenberg08 — APPROVED** [review]

**Pull request merged**